.circleci/config.yml

```diff
@@ -190,11 +191,195 @@ jobs:
+            docker push vertexproject/synapse-cortex:${CIRCLE_TAG}
+            docker push vertexproject/synapse-cryotank:${CIRCLE_TAG}
+
+  docker_branches:
+    docker:
+      - image: docker:18.05.0-ce
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: docker login
+          command: |
+            docker login --username "${DOCKER_USERNAME}" --password "${DOCKER_PASSWORD}"
+
+      - run:
+          name: build images
+          command: |
+            docker build -t vertexproject/synapse:${CIRCLE_BRANCH} -f Dockerfile .
+            docker build -t vertexproject/synapse-axon:${CIRCLE_BRANCH} -f docker/images/axon/Dockerfile .
+            docker build -t vertexproject/synapse-cortex:${CIRCLE_BRANCH} -f docker/images/cortex/Dockerfile .
+            docker build -t vertexproject/synapse-cryotank:${CIRCLE_BRANCH} -f docker/images/coryotank/Dockerfile .
+
+      - run:
+          name: tag and push container
+          command: |
+            docker push vertexproject/synapse:${CIRCLE_BRANCH}
+            docker push vertexproject/synapse-axon:${CIRCLE_BRANCH}
+            docker push vertexproject/synapse-cortex:${CIRCLE_BRANCH}
+            docker push vertexproject/synapse-cryotank:${CIRCLE_BRANCH}
+
+  docker_tags_01x:
+    docker:
+      - image: docker:18.05.0-ce
+    steps:
+      - checkout
+      - setup_remote_docker
+      - run:
+          name: docker login
+          command: |
+            docker login --username "${DOCKER_USERNAME}" --password "${DOCKER_PASSWORD}"
+      - run:
+          name: build image
+          command: |
+            docker build -t vertexproject/synapse:v0.1.x -f Dockerfile .
+            docker build -t vertexproject/synapse-axon:v0.1.x -f docker/images/axon/Dockerfile .
+            docker build -t vertexproject/synapse-cortex:v0.1.x -f docker/images/cortex/Dockerfile .
+            docker build -t vertexproject/synapse-cryotank:v0.1.x -f docker/images/coryotank/Dockerfile .
+      - run:
+          name: tag and push container
+          command: |
+            docker push vertexproject/synapse:v0.1.x
+            docker push vertexproject/synapse-axon:v0.1.x
+            docker push vertexproject/synapse-cortex:v0.1.x
+            docker push vertexproject/synapse-cryotank:v0.1.x
 
 workflows:
   version: 2
   run_tests:
     jobs:
-      - doctests
-      - osx37
-      - python37
+      - doctests:
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only: /.*/
+
+      - osx37:
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only: /.*/
+
+      - python37:
+          filters:
+            tags:
+              only: /.*/
+            branches:
+              only: /.*/
+
+      - deploy_pypi:
+          requires:
+            - doctests
+            - python37
+          filters:
+            tags:
+              only: /^v([0-9.]+)$/
```

Comment:
possible for this to match `v0`. Needs to be tighter.

Reply:
`^v([0-9]+\.[0-9]+\.[0-9]+)$`

Reply:
`^v(0\.[0-9]+\.[0-9]+)$` would prevent an accidental 1.x.y stable release version